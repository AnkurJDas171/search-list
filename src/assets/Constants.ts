export const ALERT_MESSAGE = "This user name does not exist! Please specify an existing user name!";
export const ALERT_VISIBLE_TIME = 2000;

export const LIST_TITLES = [
    'Name',
    'Rank',
    'No. of bananas',
    'isSearchedUser?'
];

export const LIST_LENGTH = 10;
