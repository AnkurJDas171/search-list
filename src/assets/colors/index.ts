const colors = {
    // UI components
    BUTTON: 'rgba(111, 202, 186, 1)',
    TEXT_INPUT: 'rgba(243, 243, 243, 1)',
    LIST_TITLE: 'rgba(111, 202, 186, 1)',
    LIST_ODD_ITEM: 'rgba(204, 255, 229, 1)',
    LIST_EVEN_ITEM: 'rgba(236, 252, 245, 1)',
    SEARCH_ITEM: 'rgba(182, 189, 186, 1)',

    // color names
    BLACK: 'rgba(0, 0, 0, 1)',
    BLACK_SEMI_TRANSPARENT: 'rgba(0, 0, 0, 0.2)',
    WHITE: 'rgba(255, 255, 255, 1)',

    // type
    TRANSPARENT: 'rgba(255, 255, 255, 0)'
}

export default colors;
