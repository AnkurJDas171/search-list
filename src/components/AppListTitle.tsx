import React from "react";
import { StyleSheet, View } from "react-native";

import AppTextContainer from "../UI/AppTextContainer";
import { LIST_TITLES } from "../assets/Constants";
import colors from "../assets/colors";

const AppListTitle = (): JSX.Element => {
    return (
        <View style={styles.body}>
            {LIST_TITLES.map((item: string) =>
                <AppTextContainer
                    title={item}
                    textColor={colors.WHITE}
                    key={item}
                />
            )}
        </View>
    )
}

const styles = StyleSheet.create({
    body: {
        alignItems: 'center',
        backgroundColor: colors.LIST_TITLE,
        display: 'flex',
        flexDirection: 'row',
        height: '10%',
        justifyContent: 'center',
    }
})

export default AppListTitle;
