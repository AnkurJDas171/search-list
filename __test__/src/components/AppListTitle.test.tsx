import React from "react";
import { render } from "@testing-library/react-native";

import AppListTitle from "../../../src/components/AppListTitle";
import { ReactTestInstance } from "react-test-renderer";
import { LIST_TITLES } from "../../../src/assets/Constants";

describe("List title component", () => {
    it("should render list title component", () => {
        const screen = render(<AppListTitle/>)
        const titles: ReactTestInstance[] = [];
        
        LIST_TITLES.forEach((title: string) => {
            titles.push(screen.getByText(title));
        })

        expect(screen.toJSON()).toMatchSnapshot();
        titles.forEach((title: ReactTestInstance) => {
            expect(title).toBeTruthy();
        })
    })
})
