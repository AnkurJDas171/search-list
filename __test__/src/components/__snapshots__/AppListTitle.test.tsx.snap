// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List title component should render list title component 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "backgroundColor": "rgba(111, 202, 186, 1)",
      "display": "flex",
      "flexDirection": "row",
      "height": 120,
      "justifyContent": "center",
    }
  }
>
  <View
    style={
      {
        "width": "25%",
      }
    }
  >
    <View
      style={
        {
          "alignItems": "center",
          "backgroundColor": "rgba(255, 255, 255, 0)",
          "borderColor": "#bcbbc1",
          "flexDirection": "row",
          "padding": 14,
        }
      }
      testID="RNE__LISTITEM__padView"
    >
      <Text
        accessibilityRole="text"
        style={
          {
            "backgroundColor": "transparent",
            "color": "rgba(255, 255, 255, 1)",
            "fontSize": 22,
            "textAlign": "left",
          }
        }
        testID="app-text-container"
      >
        Name
      </Text>
    </View>
  </View>
  <View
    style={
      {
        "width": "25%",
      }
    }
  >
    <View
      style={
        {
          "alignItems": "center",
          "backgroundColor": "rgba(255, 255, 255, 0)",
          "borderColor": "#bcbbc1",
          "flexDirection": "row",
          "padding": 14,
        }
      }
      testID="RNE__LISTITEM__padView"
    >
      <Text
        accessibilityRole="text"
        style={
          {
            "backgroundColor": "transparent",
            "color": "rgba(255, 255, 255, 1)",
            "fontSize": 22,
            "textAlign": "left",
          }
        }
        testID="app-text-container"
      >
        Rank
      </Text>
    </View>
  </View>
  <View
    style={
      {
        "width": "25%",
      }
    }
  >
    <View
      style={
        {
          "alignItems": "center",
          "backgroundColor": "rgba(255, 255, 255, 0)",
          "borderColor": "#bcbbc1",
          "flexDirection": "row",
          "padding": 14,
        }
      }
      testID="RNE__LISTITEM__padView"
    >
      <Text
        accessibilityRole="text"
        style={
          {
            "backgroundColor": "transparent",
            "color": "rgba(255, 255, 255, 1)",
            "fontSize": 22,
            "textAlign": "left",
          }
        }
        testID="app-text-container"
      >
        No. of bananas
      </Text>
    </View>
  </View>
  <View
    style={
      {
        "width": "25%",
      }
    }
  >
    <View
      style={
        {
          "alignItems": "center",
          "backgroundColor": "rgba(255, 255, 255, 0)",
          "borderColor": "#bcbbc1",
          "flexDirection": "row",
          "padding": 14,
        }
      }
      testID="RNE__LISTITEM__padView"
    >
      <Text
        accessibilityRole="text"
        style={
          {
            "backgroundColor": "transparent",
            "color": "rgba(255, 255, 255, 1)",
            "fontSize": 22,
            "textAlign": "left",
          }
        }
        testID="app-text-container"
      >
        isSearchedUser?
      </Text>
    </View>
  </View>
</View>
`;
